src/MassTransit/SystemTextJsonMessageSerializer.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace MassTransit.Serialization
+{
+    using MassTransit.Metadata;
+    using MassTransit.Util;
+    using System;
+    using System.IO;
+    using System.Net.Mime;
+    using System.Runtime.Serialization;
+    using System.Text.Json;
+
+    public class SystemTextJsonMessageSerializer : IMessageSerializer
+    {
+        public ContentType ContentType => JsonContentType;
+
+        internal static readonly ContentType JsonContentType = new ContentType("application/vnd.masstransit+json2");
+
+        public void Serialize<T>(Stream stream, SendContext<T> context) where T : class
+        {
+            try
+            {
+                context.ContentType = JsonContentType;
+
+                var envelope = new JsonMessageEnvelope(context, context.Message, TypeMetadataCache<T>.MessageTypeNames);
+                AsyncUtil.RunSync(() => JsonSerializer.SerializeAsync<MessageEnvelope>(stream, envelope, SystemTextJsonConfiguration.Options));
```

Comment:
so this isn't going to fly with me, and will need to be deferred until the serializer interfaces are changed to async.

Reply:
when it is going happen ?

Reply:
Maybe in the next month or two.